source/SpinalHDL/Semantic/when_switch.rst

```diff
@@ -97,6 +97,65 @@ is equivalent to
     }
   }
 
+
+Additional options
+^^^^^^^^^^^^^^^^^^
+
+Sometimes handling all cases can become unwieldy and error prone so SpinalHDL, by default, will handle the uncovered cases with the last ``is`` block.
+To explicitly declare and define a ``default`` block the option ``coverUnreachable`` can be passed to the switch
+
+.. code-block:: scala
+
+  switch(aluop, coverUnreachable = true) {
+    is(ALUOp.add, ALUOp.slt, ALUOp.sltu) {
+        immediate := instruction.immI.signExtend
+    }
+    is(ALUOp.sll, ALUOp.sra) {
+        immediate := instruction.shamt
+    }
+    default{
+        immediate := 0
+    }
+  }
+
+If the used values of ``ALUOp`` are all available elements of the SpinalEnum type, then without the ``coverUnreachable = true`` option SpinalHDL would throw an UNREACHABLE STATEMENT error upon elaboration.
+Without the ``default`` block the encoding for ``ALUOp.sll`` and ``ALUOp.sra`` are the default cases in the generated HDL.
+
+When using defined constants to compare against within a ``switch`` block it can occasionally happen that duplicates within a ``is`` value occur. 
+By default duplicates of ``is`` conditions are not ignored but identified as an error. 
+To relax the strictness of the ``switch`` elaboration the ``strict = false`` can be passed (by default ``strict = true`` thus preventing duplicate ``is`` conditions).
+
+.. code-block:: scala
```

Comment:
I do not understand the example, can you explain the consequence of using `strict = false` here, please? (Behavior with vs without it)

Reply:
So if I leave out `strict = false` what happens is that SpinalHDL gives an error (`DUPLICATED ELEMENTS IN SWITCH IS(...) STATEMENT`). Using this option relaxes the elaboration of the switch block to allow the duplicate element (and only keep one in the generated Verilog (as otherwise Verilator would complain too -- I remember talking about this to Dolu when I got this issue the first time, I dont recall if I created an issue or just asked in the gitter).

Reply:
So which one is used? The last one (last-assignment-wins rule)?

I suggest this kind of structure, which is more concise and explains the example.

```rst
By default, duplicates of ``is`` conditions are identified as an error.
This strictness can be released passing ``strict = false`` to ``switch``, then XXX happens.
In the example below, Xxx and Xxx are duplicates, and ``strict = false`` makes XXX happen when XXX.
```